go-app-ussd_popi_faq.js

```diff
@@ -63,32 +61,7 @@ go.app = function() {
             ;
         };
 
-        self.add = function(name, creator) {
-            self.states.add(name, function(name, opts) {
-                if (!interrupt || !utils.timed_out(self.im))
-                    return creator(name, opts);
-
-                interrupt = false;
-                var timeout_opts = opts || {};
-                timeout_opts.name = name;
-                return self.states.create('state_timed_out', timeout_opts);
-            });
-        };
-
-        self.states.add('state_timed_out', function(name, creator_opts) {
-            return new ChoiceState(name, {
-                question: $('Welcome back. Please select an option:'),
-                choices: [
-                    new Choice(creator_opts.name, $('Continue viewing FAQ')),
-                    new Choice('state_start', $('Main menu'))
-                ],
-                next: function(choice) {
-                    return choice.value;
-                }
-            });
-        });
```

Comment:
Removing the timeout states means that they'll go back to whatever state they were when they timed out.

So if they were busy viewing a long screen of information, and then they timeout, and then redial in again, they'll be shown that long screen as the first thing.

The idea with the timeout state is that when they redial, they'll be shown that timeout state instead, which we make sure is <140 chars. They then click continue, and see whatever screen they're on. That way we ensure that the first screen of every session is <140 chars.